openpype/plugins/publish/extract_color_transcode.py

```diff
@@ -0,0 +1,212 @@
+                    if not new_repre.get("custom_tags"):
+                        new_repre["custom_tags"] = []
+                    new_repre["custom_tags"].extend(custom_tags)
+
+                # Add additional tags from output definition to representation
+                for tag in output_def["tags"]:
+                    if tag not in new_repre["tags"]:
+                        new_repre["tags"].append(tag)
+
+                instance.data["representations"].append(new_repre)
+
+    def _rename_output_files(self, files_to_convert, output_extension):
+        """Change extension of converted files."""
+        renamed_files = []
+        for file_name in files_to_convert:
+            file_name, _ = os.path.splitext(file_name)
+            new_file_name = '{}.{}'.format(file_name,
+                                           output_extension)
+            renamed_files.append(new_file_name)
+        files_to_convert = renamed_files
+        return files_to_convert
+
+    def _get_profile(self, instance):
+        """Returns profile if and how repre should be color transcoded."""
+        host_name = instance.context.data["hostName"]
+        family = instance.data["family"]
+        task_data = instance.data["anatomyData"].get("task", {})
+        task_name = task_data.get("name")
+        task_type = task_data.get("type")
+        subset = instance.data["subset"]
+        filtering_criteria = {
+            "hosts": host_name,
+            "families": family,
+            "task_names": task_name,
+            "task_types": task_type,
+            "subset": subset
+        }
+        profile = filter_profiles(self.profiles, filtering_criteria,
+                                  logger=self.log)
+
+        if not profile:
+            self.log.info((
+                  "Skipped instance. None of profiles in presets are for"
+                  " Host: \"{}\" | Families: \"{}\" | Task \"{}\""
+                  " | Task type \"{}\" | Subset \"{}\" "
+              ).format(host_name, family, task_name, task_type, subset))
+
+        self.log.debug("profile: {}".format(profile))
+        return profile
+
+    def _repre_is_valid(self, repre):
+        """Validation if representation should be processed.
+
+        Args:
+            repre (dict): Representation which should be checked.
+
+        Returns:
+            bool: False if can't be processed else True.
+        """
+
+        if repre.get("ext") not in self.supported_exts:
+            self.log.debug((
+                    "Representation \"{}\" of unsupported extension. Skipped."
+            ).format(repre["name"]))
+            return False
+
+        if not repre.get("files"):
+            self.log.debug((
+                "Representation \"{}\" have empty files. Skipped."
+            ).format(repre["name"]))
+            return False
+
+        if not repre.get("colorspaceData"):
+            self.log.debug("Repre has no colorspace data. Skipped.")
```

Comment:
This log should also mention the name of the representation like the above - plus I wouldn't abbreviate `Representation` in the logged message to `Repre`.

Reply:
Makes sense.